docs/running_psalm/plugins/plugins_type_system.md

```diff
@@ -41,16 +41,40 @@ The classes are as follows:
 
 `TResource` - denotes the `resource` type (e.g. a file handle).
 
+`TClosedResource` - denotes the `resource` type that has been closed (e.g. a file handle through `fclose()`).
+
+`TAssertionFalsy` - Represents any value reduced to false when computed in boolean context. This is used for assertions
+
+`TConditional` - Internal representation of a conditional return type in phpdoc. For example ($param1 is int ? int : string)
+
+`TIntMask` - Represents the type that is the result of a bitmask combination of its parameters. `int-mask<1, 2, 4>` corresponds to `1|2|3|4|5|6|7`
+
+`TIntMaskOf` - as above, but used with with a reference to constants in code`int-mask<MyClass::CLASS_CONSTANT_*>` will corresponds to `1|2|3|4|5|6|7` if there are three constant 1, 2 and 4
+
+`TKeyOfClassConstant` - Represents an offset of a class constant array.
+
+`TTemplateIndexedAccess` - To be documented
```

Comment:
My guess would be that it corresponds to `T[K]` type mentioned in https://psalm.dev/docs/annotating_code/type_syntax/atomic_types/#magical-types

Reply:
mmh, not much description on that one. Not sure what I should add

Reply:
See release notes for [Psalm 3.3.1](https://github.com/vimeo/psalm/releases/tag/3.3.1)

Reply:
Well, I added 
> Represents the type used when using TKeyOfClassConstant when the type of the class constant array is a template

Not sure I would be able to tell what it is if I didn't just checked the example but well, better than nothing

Reply:
Typescript docs may provide some inspiration: https://www.typescriptlang.org/docs/handbook/advanced-types.html#index-types